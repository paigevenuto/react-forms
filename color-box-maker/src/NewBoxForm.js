import React, { useState } from "react";

function NewBoxForm({ addBox }) {
  const [formData, setFormData] = useState({
    backgroundColor: "",
    width: "",
    height: "",
  });

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addBox(formData);
    setFormData({
      backgroundColor: "",
      width: "",
      height: "",
    });
  };

  return (
    <form className="NewBoxForm" onSubmit={handleSubmit}>
      <label htmlFor="backgroundColor">Box Color:</label>
      <input
        list="backgroundColors"
        name="backgroundColor"
        value={formData.backgroundColor}
        onChange={handleChange}
        required
      />
      <datalist id="backgroundColors">
        <option value="red" />
        <option value="blue" />
        <option value="green" />
        <option value="pink" />
        <option value="teal" />
        <option value="purple" />
        <option value="orange" />
        <option value="yellow" />
        <option value="magenta" />
      </datalist>
      <label htmlFor="width">Box Width:</label>
      <input
        type="number"
        name="width"
        value={formData.width}
        onChange={handleChange}
        required
      />
      <label htmlFor="height">Box Height:</label>
      <input
        type="number"
        name="height"
        value={formData.height}
        onChange={handleChange}
        required
      />
      <button>Add!</button>
    </form>
  );
}

export default NewBoxForm;
