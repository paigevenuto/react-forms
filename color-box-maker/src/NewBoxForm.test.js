import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

// smoke test
it("renders without crashing", function () {
  render(<NewBoxForm />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
