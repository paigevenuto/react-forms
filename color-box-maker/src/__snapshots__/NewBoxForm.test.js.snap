// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="backgroundColor"
    >
      Box Color:
    </label>
    <input
      id="backgroundColor"
      list="backgroundColors"
      name="backgroundColor"
      required=""
      value=""
    />
    <datalist
      id="backgroundColors"
    >
      <option
        value="red"
      />
      <option
        value="blue"
      />
      <option
        value="green"
      />
      <option
        value="pink"
      />
      <option
        value="teal"
      />
      <option
        value="purple"
      />
      <option
        value="orange"
      />
      <option
        value="yellow"
      />
      <option
        value="magenta"
      />
    </datalist>
    <label
      for="width"
    >
      Box Width:
    </label>
    <input
      id="width"
      name="width"
      required=""
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Box Height:
    </label>
    <input
      id="height"
      name="height"
      required=""
      type="number"
      value=""
    />
    <button>
      Add!
    </button>
  </form>
</DocumentFragment>
`;
